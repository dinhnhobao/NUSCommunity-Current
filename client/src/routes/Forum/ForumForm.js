import React from "react";
import {withRouter} from "react-router-dom";
import BraftEditor from "braft-editor";
import dayjs from "dayjs";
import { Form, Input, message } from "antd";
import { PlusCircleFilled, CloseOutlined } from "@ant-design/icons";
import "braft-editor/dist/index.css";
import "./index.css";

const controls = ["bold", "italic", "headings", "text-color", "emoji"];

class ForumForm extends React.Component {
  formRef = React.createRef();

  constructor(props) {
    super(props)
    this.state = {
      tags: [],
      tag: "",
      email:"",
      username: "",
      bio:""
    };
  }

  componentDidMount() {
    const { data } = this.props;
    if (data.id) {
      this.setState({
        tags: data.tags,
      });
    }
    this.getUserData();
  }

  getUserData() {
    fetch('/auth/check-session', {
      credentials: 'include'
    })
    .then(res => res.json())
    .then(response => {
      if (!response.loggedin) {
        message.error('Please register and login first to access!')
        this.props.history.push('/')
      }
      this.setState({
        email: response.email,
        username: response.username,
        bio: response.bio
      })
    })
  }

  onFinish = (values) => {
    const { tags } = this.state;
    const { onSubmit, data } = this.props;
    if (!tags.length) {
      message.error("please enter your tag name");
      return;
    }
    const params = {
      id: data.id ? data.id : dayjs().valueOf(),
      title: values.title,
      msg: values.msg.toHTML(),
      email: this.state.email,
      username: this.state.username,
      bio: this.state.bio,
      time:dayjs().valueOf(),
      //time: dayjs.format('YYYY-MM-DD hh:mm:ss'),
      tags: this.state.tags
    };
    onSubmit(params);
  };

  addTag = () => {
    const { tags, tag } = this.state;
    if (!tag) {
      message.error("please enter your tag name");
      return;
    }
    this.setState({
      tags: [...tags, tag],
      tag: "",
    });
  };

  removeTag = (name) => {
    const { tags } = this.state;
    const data = [];
    tags.forEach((item) => {
      if (item !== name) {
        data.push(item);
      }
    });
    this.setState({
      tags: data,
    });
  };

  remove = () => {
    const { data, history } = this.props;
    const forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];
    //message.success("delete success~");
    sessionStorage.setItem(
      "forumList",
      JSON.stringify(
        forumList.filter((item) => {
          return Number(item.id) !== Number(data.id);
        })
      )
    );

    fetch('/forum/delete-post', {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then(res => res.json())
    .then(res => {
      console.log(res.message);
      if (res.message === "Success") {
        message.success("Delete post successfully!")
      } else {
        message.error("Fail to delete post!")
      }
      history.push("/index/forum");
    })
  };

  render() {
    const { data } = this.props;
    const { tags, tag } = this.state;
    return (
      <div className="forumForm">
        <Form
          name="normal_add"
          ref={this.formRef}
          className="forumForm-form"
          onFinish={this.onFinish}
          initialValues={{
            title: data.title,
            msg: BraftEditor.createEditorState(data.msg || ""),
          }}
        >
          <Form.Item
            name="title"
            rules={[
              {
                required: true,
                message: "please enter your title!",
              },
            ]}
          >
            <Input placeholder="Discussion title" />
          </Form.Item>
          <Form.Item>
            <span className="forumForm-form-tag-msg">Tagsï¼š</span>
            {tags.map((item) => (
              <div key={item} className="forumForm-form-tag-item">
                {item}
                <CloseOutlined
                  onClick={() => this.removeTag(item)}
                  className="forumForm-form-tag-item-remove"
                />
              </div>
            ))}
            <Input
              value={tag}
              onChange={(e) => this.setState({ tag: e.target.value })}
              className="forumForm-form-tag-input"
              placeholder="tag name..."
            />
            <PlusCircleFilled
              onClick={() => this.addTag()}
              className="forumForm-form-tag-icon"
            />
          </Form.Item>
          <Form.Item
            name="msg"
            validateTrigger="onBlur"
            rules={[
              {
                validator(rule, value) {
                  if (value.isEmpty()) {
                    return Promise.reject("please enter your summary");
                  }
                  return Promise.resolve();
                },
              },
            ]}
          >
            <BraftEditor
              language="en"
              className="my-editor"
              controls={controls}
              placeholder="Discussion summary..."
            />
          </Form.Item>
          <div
            onClick={() => {
              this.formRef.current.submit();
            }}
            className={data.id ? "forumForm-form-btn" : "forumForm-form-btn-2"}
          >
            Post
          </div>
          {data.id && (
            <div
              onClick={() => {
                this.remove();
              }}
              className="forumForm-form-btn-del"
            >
              Delete
            </div>
          )}
        </Form>
      </div>
    );
  }
}

export default withRouter(ForumForm);