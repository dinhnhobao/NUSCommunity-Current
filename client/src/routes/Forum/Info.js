import React from "react";
import dayjs from "dayjs";
import { message, Form } from "antd";
import BraftEditor from "braft-editor";
import {
  UserAddOutlined,
  DeleteOutlined,
  LikeOutlined,
  // HeartOutlined,
} from "@ant-design/icons";
import { parse } from "qs";
import "braft-editor/dist/index.css";

const controls = ["bold", "italic", "headings", "text-color", "emoji"];

export default class Info extends React.Component {
  formRef = React.createRef();

  state = {
    data: {},
    comment: [
      {
        comment_web_id: dayjs().valueOf(),
        time: dayjs().valueOf(),
        username: "abc",
        value: "<div>111</div>",
      },
    ],
    like: false,
    favorite: false,
    userComment:'',
  };

  componentDidMount() {
    const query = this.getPageQuery();
    const forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];
    this.setState({
      data: forumList.find((item) => {
        return Number(item.id) === Number(query.id);
      }),
    });
    setTimeout(() => {
      this.formRef.current.setFieldsValue({
        comment: BraftEditor.createEditorState(""),
      });
    }, 100);
    this.getUserComment();
  }

  getUserComment() {
    fetch('/auth/check-session', {
      credentials: 'include'
    })
    .then(res => res.json())
    .then(response => {
      this.state.userComment = response.username;
    })
  }

  getPageQuery = () => parse(window.location.href.split("?")[1]);

  remove = () => {
    const { history } = this.props;
    const query = this.getPageQuery();
    const forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];
    message.success("delete success~");
    sessionStorage.setItem(
      "forumList",
      JSON.stringify(
        forumList.filter((item) => {
          return Number(item.id) !== Number(query.id);
        })
      )
    );
    history.push("/index/forum");
  };

  onFinish = (values) => {
    const { comment } = this.state;
    this.formRef.current.setFieldsValue();

    const comment_web_id = dayjs().valueOf();
    const time = dayjs().valueOf();
    
    this.setState({
      comment: [
        ...comment,
        {
          comment_web_id: comment_web_id,
          time: time,
          username: this.state.userComment,
          value: values.comment.toHTML(),
        },
      ],
    });

    const params = {
      post_web_id: this.state.data.id,
      comment_web_id: comment_web_id,
      username: this.state.userComment,
      value: values.comment.toHTML(),
      time: time,
    } 

    fetch('/forum/add-comment', {
      method: 'POST',
      body: JSON.stringify(params),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then(res => res.json())
    .then(res => {
      console.log(res.message);
      if (res.message == "Success") {
        message.success("Add comment successfully!")
      } else {
        message.error("Fail to add comment!")
      }
    })

    this.formRef.current.setFieldsValue({
      comment: BraftEditor.createEditorState(""),
    });
  };

  removeComment = (id) => {
    const { comment } = this.state;
    //message.success("delete success~");

    const post_web_id = this.state.data.id;
    const params = {
      post_web_id: post_web_id,
      comment_web_id: id,
    }

    if (this.state.data.username === this.state.userComment) {
      this.setState({
        comment: comment.filter((item) => {
          return Number(item.comment_web_id) !== id;
        }),
      });

      fetch('/forum/delete-comment', {
        method: 'POST',
        body: JSON.stringify(params),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      .then(res => res.json())
      .then(res => {
          console.log(res.message);
          if (res.message == "Success") {
            message.success("Delete comment successfully!")
          } else {
            message.error("Fail to delete post!")
          }
      })
    } else {
      message.error("You are not allowed to delete comment!");
    }
  };

  render() {
    const { data, favorite, like, comment } = this.state;
    if (!data.id) {
      return <div />;
    }
    return (
      <div className="forumInfo">
        <div className="forumInfo-data">
          <div className="forumInfo-data-user">
            <UserAddOutlined className="mr-8" />
            {data.username}
            <div className="forumInfo-data-user-time">
              {dayjs(data.time).fromNow()}
            </div>
          </div>
          <div className="forumInfo-data-title">{data.title}</div>
          <div
            className="forumInfo-data-msg"
            dangerouslySetInnerHTML={{ __html: data.msg }}
          />
          <div className="forumInfo-data-bottom">
            {data.tags.map((res) => (
              <div key={res} className="forumInfo-data-bottom-tag">
                {res}
              </div>
            ))}
            <div className="forumInfo-data-bottom-blank" />
            <LikeOutlined
              onClick={() => {
                this.setState({ like: !like });
              }}
              className={`forumInfo-data-bottom-icon ${like ? "red" : ""}`}
            />
            {/* <HeartOutlined
              onClick={() => {
                this.setState({ favorite: !favorite });
              }}
              className={`forumInfo-data-bottom-icon ${favorite ? "red" : ""}`}
            />
            <DeleteOutlined
              onClick={this.remove}
              className="forumInfo-data-bottom-icon"
            /> */}
          </div>
        </div>
        <div className="forumInfo-comment">
          <Form
            name="normal_comment"
            ref={this.formRef}
            className="forumInfo-comment-form"
            onFinish={this.onFinish}
          >
            <Form.Item
              name="comment"
              validateTrigger="onBlur"
              rules={[
                {
                  validator(rule, value) {
                    if (value.isEmpty()) {
                      return Promise.reject("please enter your opinion");
                    }
                    return Promise.resolve();
                  },
                },
              ]}
            >
              <BraftEditor
                language="en"
                className="my-editor"
                controls={controls}
                placeholder="Your opinion..."
              />
            </Form.Item>
            <div
              onClick={() => {
                this.formRef.current.submit();
              }}
              className="forumInfo-comment-form-btn"
            >
              Reply
            </div>
          </Form>
        </div>
        <div className="forumInfo-commentList">
          {comment.map((item) => (
            <div key={item.comment_web_id} className="forumInfo-commentList-item">
              <div className="forumInfo-commentList-item-header">
                <div className="forumInfo-commentList-item-header-user">
                  <UserAddOutlined className="mr-8" />
                  {item.username}
                </div>
                <div className="forumInfo-commentList-item-header-time">
                  {dayjs(item.time).fromNow()}
                </div>
              </div>
              <div
                className="forumInfo-commentList-item-con"
                dangerouslySetInnerHTML={{ __html: item.value }}
              />
              <DeleteOutlined
                onClick={() => this.removeComment(item.comment_web_id)}
                className="forumInfo-commentList-item-del"
              />
            </div>
          ))}
        </div>
      </div>
    );
  }
}
