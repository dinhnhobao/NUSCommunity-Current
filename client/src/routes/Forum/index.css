.forum {
  display: flex;
}
.forum-add {
  width: 260px;
}
.forum-add-form {
  margin-left: 10px;
  margin-bottom: 10px;
}
.forum-add-form .ant-input {
  width: 200px;
}
.forum-add-form-search {
  margin-right: 0 !important;
}
.forum-add-btn {
  margin-left: 10px;
  color: #999;
  border: 1px solid #e8e8e8;
  line-height: 38px;
  display: block;
  text-align: center;
  font-weight: bolder;
}
.forum-list {
  flex: 1;
}
.forum-list-title {
  border: 1px solid #e8e8e8;
  height: 30px;
  padding-left: 10px;
  font-weight: bolder;
  line-height: 28px;
}
.forum-list-item {
  border: 1px solid #e8e8e8;
  border-top: none;
  padding: 10px;
}
.forum-list-item-title {
  font-size: 18px;
  line-height: 22px;
  font-weight: 300;
  color: #000;
}
.forum-list-item-user {
  margin-top: 10px;
  line-height: 22px;
  font-weight: 700;
  color: #999;
}
.forum-list-item-bottom {
  margin-top: 10px;
  font-size: 12px;
  display: flex;
}
.forum-list-item-bottom-tag {
  padding: 0 10px;
  background-color: #f9f9f9;
  color: #999;
  line-height: 24px;
  margin-right: 8px;
}
.forum-list-item-bottom-edit {
  line-height: 24px;
  color: #999;
  flex: 1;
  text-align: right;
  margin-right: 16px;
}
.forum-list-item-bottom-time {
  font-weight: 700;
  line-height: 24px;
  color: #999;
  text-align: right;
}

.forumForm {
  display: flex;
}
.forumForm-form {
  width: 100%;
  position: relative;
}
.forumForm-form .my-editor {
  border: 1px solid #d9d9d9;
  padding-bottom: 64px;
}
.forumForm-form-btn {
  position: absolute;
  border-top: 1px solid #d9d9d9;
  bottom: 56px;
  width: 100%;
  line-height: 32px;
  text-align: center;
  cursor: pointer;
  font-weight: 600;
}

.forumForm-form-btn-2 {
  position: absolute;
  border-top: 1px solid #d9d9d9;
  bottom: 24px;
  width: 100%;
  line-height: 32px;
  text-align: center;
  cursor: pointer;
  font-weight: 600;
}

.forumForm-form-btn-del {
  position: absolute;
  border-top: 1px solid #d9d9d9;
  bottom: 24px;
  width: 100%;
  line-height: 32px;
  text-align: center;
  cursor: pointer;
  font-weight: 600;
  color: red;
}
.forumForm-form-btn:hover {
  color: #000;
}
.forumForm-form-tag {
  display: flex;
}
.forumForm-form-tag-msg {
  color: #000;
  font-weight: 600;
}
.forumForm-form-tag-item {
  padding: 0 10px;
  background-color: #f9f9f9;
  color: #999;
  line-height: 24px;
  display: inline-block;
  margin: 0 5px;
}
.forumForm-form-tag-item-remove {
  margin-left: 5px;
  cursor: pointer;
}
.forumForm-form-tag-input {
  width: 200px;
  border: none;
}
.forumForm-form-tag-input:focus {
  border: none;
  box-shadow: none;
}
.forumForm-form-tag-icon {
  font-size: 20px;
  cursor: pointer;
}

.forumInfo-data {
  padding: 20px;
}
.forumInfo-data-user {
  line-height: 22px;
  font-weight: 700;
  color: #999;
  display: flex;
}
.forumInfo-data-user-time {
  font-weight: 700;
  line-height: 24px;
  color: #999;
  text-align: right;
  flex: 1;
}
.forumInfo-data-title {
  margin-top: 15px;
  font-size: 24px;
  line-height: 32px;
  font-weight: 400;
  color: #000;
}
.forumInfo-data-msg {
  margin-top: 10px;
  line-height: 26px;
}
.forumInfo-data-bottom {
  margin-top: 10px;
  font-size: 12px;
  display: flex;
}
.forumInfo-data-bottom-tag {
  padding: 0 10px;
  background-color: #f9f9f9;
  color: #999;
  line-height: 24px;
  margin-right: 8px;
}
.forumInfo-data-bottom-blank {
  flex: 1;
}
.forumInfo-data-bottom-icon {
  font-size: 16px;
  margin-right: 16px;
  cursor: pointer;
}
.forumInfo-comment {
  display: flex;
}
.forumInfo-comment-form {
  width: 100%;
  position: relative;
}
.forumInfo-comment-form .my-editor {
  border: 1px solid #e8e8e8;
  height: 132px;
  padding-bottom: 32px;
}
.forumInfo-comment-form .bf-content {
  height: 60px;
  padding-bottom: 0;
}
.forumInfo-comment-form-btn {
  width: 100%;
  line-height: 32px;
  text-align: center;
  cursor: pointer;
  font-weight: 600;
  border-top: 1px solid #e8e8e8;
  position: absolute;
  bottom: 24px;
}

.forumInfo-commentList-item {
  padding: 0 8px;
  border: 1px solid #e8e8e8;
  margin-bottom: 16px;
  padding-bottom: 30px;
  position: relative;
}

.forumInfo-commentList-item-header {
  font-size: 16px;
  border-bottom: 1px solid #e8e8e8;
  display: flex;
  line-height: 40px;
}

.forumInfo-commentList-item-header-user {
  flex: 1;
}

.forumInfo-commentList-item-con {
  padding: 10px 0;
}

.forumInfo-commentList-item-del {
  font-size: 16px;
  position: absolute;
  bottom: 8px;
  right: 8px;
}
