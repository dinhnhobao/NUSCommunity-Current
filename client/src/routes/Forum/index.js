import React from "react";
import { Link } from "react-router-dom";
import dayjs from "dayjs";
import { Empty, Form, Input, Button, Modal, Tag } from "antd";
import {
  UserAddOutlined,
  SearchOutlined,
  EditOutlined,
} from "@ant-design/icons";
import "./index.css";

const relativeTime = require("dayjs/plugin/relativeTime");
dayjs.extend(relativeTime);

const colors = ["magenta", "orange", "cyan", "purple"];

export default class Forum extends React.Component {
  state = {
    visible: false,
    data: { username: "", email: "", bio: "" },
    searchKey:"",
  };
  onFinish = (values) => {
    const {history} = this.props;
    console.log(values);
    this.setState({
      searchKey: values.discussion,
    })

    history.push("/index/forum");
  };

  render() {
    var forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];

    if (this.state.searchKey !== ''){
     forumList = forumList.filter((item) => item.tags.map(x => x.toLowerCase()).includes(this.state.searchKey.toLowerCase()));
    }
    
    const { visible, data } = this.state;
    const layout = {
      labelCol: {
        span: 4,
      },
      wrapperCol: {
        span: 16,
      },
    };
    return (
      <div className="forum">
        <div className="forum-list">
          <div className="forum-list-title">DISCUSSIONS</div>
          {forumList.map((item) => (
            data.username = item.username, data.email = item.email, data.bio = item.bio,
            <div key={item.time} className="forum-list-item">
              <Link
                to={`/index/forum/info?id=${item.id}`}
                className="forum-list-item-title"
              >
                {item.title}
              </Link>
              <div className="forum-list-item-user">
                <UserAddOutlined className="mr-8 blue" />
                <span
                  onClick={() => {
                    this.setState({ visible: true });
                  }}
                  style={{ cursor: "pointer" }}
                >
                  {item.username}
                </span>
              </div>
              <div className="forum-list-item-bottom">
                {item.tags.map((res, index) => (
                  <Tag key={res} color={colors[index % 4]}>
                    {res}
                  </Tag>
                ))}
                <Link
                  to={`/index/forum/edit?id=${item.id}`}
                  className="forum-list-item-bottom-edit"
                >
                  <EditOutlined /> edit
                </Link>
                <div className="forum-list-item-bottom-time">
                  {dayjs(item.time).fromNow()}
                </div>
              </div>
            </div>
          ))}
          {forumList.length ? null : (
            <Empty style={{ margin: 0 }} className="forum-list-item" />
          )}
        </div>
        <div className="forum-add">
          <Form
            name="horizontal_forum"
            layout="inline"
            onFinish={this.onFinish}
            className="forum-add-form"
          >
            <Form.Item name="discussion">
              <Input placeholder="search discussion" />
            </Form.Item>
            <Form.Item className="forum-add-form-search">
              <Button
                type="primary"
                htmlType="submit"
                className="forum-button-search"
                icon={<SearchOutlined />}
              />
            </Form.Item>
          </Form>
          <Button
            type="primary"
            href="/index/forum/add"
            className="forum-add-btn"
          >
            New Discussion
          </Button>
        </div>
        <Modal
          visible={visible}
          title="User Info"
          className="modal-user-display"
          onCancel={() => {
            this.setState({ visible: false });
          }}
          footer={null}
        >
          <Form {...layout}>
            <Form.Item label="Name">{data.username}</Form.Item>
            <Form.Item label="Email">{data.email}</Form.Item>
            <Form.Item label="Bio">{data.bio}</Form.Item>
          </Form>
        </Modal>
      </div>
    );
  }
}
