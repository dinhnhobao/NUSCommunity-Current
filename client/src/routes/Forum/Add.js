import React from "react";
import { message } from "antd";
import ForumForm from "./ForumForm";

export default class Add extends React.Component {
  onSubmit = (values) => {
    const { history } = this.props;
    const forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];
    message.success("add success~");
    sessionStorage.setItem("forumList", JSON.stringify([...forumList, values]));
    history.push("/index/forum");
  };

  render() {
    return <ForumForm data={{}} onSubmit={this.onSubmit} />;
  }
}
