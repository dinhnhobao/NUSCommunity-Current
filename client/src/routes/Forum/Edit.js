import React from "react";
import { message } from "antd";
import { parse } from "qs";
import ForumForm from "./ForumForm";

export default class Edit extends React.Component {
  state = {
    data: {},
  };
  componentDidMount() {
    const query = this.getPageQuery();
    const forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];
    this.setState({
      data: forumList.find((item) => {
        return Number(item.id) === Number(query.id);
      }),
    });
  }
  getPageQuery = () => parse(window.location.href.split("?")[1]);

  onSubmit = (values) => {
    const { history } = this.props;
    const forumList = sessionStorage.getItem("forumList")
      ? JSON.parse(sessionStorage.getItem("forumList"))
      : [];
    const list = [];
    forumList.forEach((item) => {
      if (Number(item.id) === Number(values.id)) {
        list.push(values);
      } else {
        list.push(item);
      }
    });
    message.success("edit success~");
    sessionStorage.setItem("forumList", JSON.stringify(list));
    history.push("/index/forum");
  };

  render() {
    const { data } = this.state;
    //console.log(data);
    if (!data.id) {
      return <div />;
    }
    return <ForumForm data={data} onSubmit={this.onSubmit} {...this.props} />;
  }
}
