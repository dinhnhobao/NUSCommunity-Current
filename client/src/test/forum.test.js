import React from 'react';
import {shallow} from 'enzyme';
import Forum from '../routes/Forum/index';

const setUp = (props={}) => {
    const component = shallow(<Forum {...props} />);
    return component;
}

const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}

describe('Forum Component', () => {
    let component;
    beforeEach(() => {
        component = setUp();
    })

    it('Should render correctly', () => {
        const wrapper =  findByTestAttr(component, 'forumComponent');
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const wrapper =  findByTestAttr(component, 'forumComponent');
        expect(wrapper.length).toEqual(1);       
    });

    describe('Form List', () => {
        it('Should render forum list', () => {
            const wrapper =  findByTestAttr(component, 'forumList');
            expect(wrapper).toMatchSnapshot();
        });
    })

    describe('Search Button', () => {
        it('Should render Search Button', () => {
            const wrapper =  findByTestAttr(component, 'searchButtonComponent');
            expect(wrapper.length).toEqual(1);
        })
    })
})
