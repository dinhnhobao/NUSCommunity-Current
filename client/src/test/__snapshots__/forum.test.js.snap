// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forum Component Form List Should render forum list 1`] = `ShallowWrapper {}`;

exports[`Forum Component Should render correctly 1`] = `ShallowWrapper {}`;
